### DISCUSSION TITLE
Shrink on load values rationale

### DISCUSSION THREAD

**Discussion by awesomecoder111** [libvips/libvips#2200] [General]
Hi @jcupitt!

Consider we have input jpg image that's 400x400 and I'm generating a thumbnail that's 100x100. My understanding of vips thumbnail is that it'll shrink the image while loading to 200x200 and then use lanczos to downsize to 100x100 based off the shrink value here:
https://github.com/libvips/libvips/blob/da6fe210b4c045edabe4799c5db5d3fc25e7876e/libvips/resample/thumbnail.c#L497

I have three questions:

1. Ideally, I'd like to load a 141x141 image using shrink on load since the number of pixels in this image is 2x that in the 100x100 image. This isn't possible to do with libjpeg right since the scalar shrink factor is applied to each dimension?

2. Some of your comments online imply that we need to make sure that image loaded after libjpeg shrink needs to be twice the size of the target thumbnail for a good quality image. Are there any experiments showing this?  

3. I'm wondering if I had a 300x300 image and was generating a 100x100 thumbnail, could i apply a shrink factor of 2 to get a 150x150 image and get decent thumbnails? I tested it out and couldn't find any discernable difference in the thumbnails, but i'm wondering if there's some images for which this wouldn't look good.

**Comment by jcupitt** [comment]
Hi @awesomecoder111,

Yes, libjpeg shrink-on-load is a block filter (imagine a square wave). Suppose you have 100 equal values side by side -- what's the widest gaussian curve you can fit in that 100 wide block? It'll be about 50 wide at the waist.

There are some useful test images here:

http://www.rollthepotato.net/~john/kevill9/

Try shrinking this 800x600 jpg image in various ways. The roof will produce terrible moire fringes if you get it wrong:

http://www.rollthepotato.net/~john/kevill9/test_800x600.jpg

For example, this is bad:

http://www.rollthepotato.net/~john/kevill/vips-400x300.png

And this is OK:

http://www.rollthepotato.net/~john/kevill9/400x300-vips.png

Check the vertical edges on the windows for sharpness.

**Comment by jcupitt** [comment]
Here's a really difficult test case:

http://www.rollthepotato.net/~john/IMG_1869.JPG

It's a photo I took at the Olympic Park in East London. The roof is made of thin wooden slats and is hard to resize well.
